### ISSUE TITLE
2023 Technical Debt Cleanup

### ISSUE THREAD

**Issue by Purg** [PTG-Kitware/angel_system#373]
This is intended to be a somewhat-living list of technical debt we've recently accrued. Feel free to comment with discussion or additional items that could be added to this list.

- [ ] Update ROS2 backing away from Foxy (deprecated) and to the latest version if possible (Iron)
- [ ] With the movement over to the Yolo v7 object detector, I think we can deprecate/remove the Detectron2 "clone" that had previously become embedded within our python package: `./angel_system/berkeley/` is literally a complete, embedded fork of the detectron2. If want to return to this in the future, this needs to be integrated differently...

**Comment by hdefazio** [comment]
I'm currently working on removing a bunch of old and unused stuff and some slight reorganization

**Comment by cameron-a-johnson** [comment]
- [ ] Assure that new activity configs and task configs can be swapped into the GSP with minimal friction. Test on the tourniquet-wrapping configs.

**Comment by Purg** [comment]
@joshanderson-kw Did we move away from using the python package trimesh? There are a number of system packages we're installing that we added for that.

**Comment by hdefazio** [comment]
- [ ] Clean up python dependencies 
- [ ] Move the yolov7 repo to the PTG space

**Comment by hdefazio** [comment]
Initial removal of old parts is here: https://github.com/PTG-Kitware/angel_system/pull/374, this has been tested with a ros bag, I don't have a HL here to test it live

**Comment by Purg** [comment]
Where do other remaining items here stand w.r.t. being technical debt as opposed to feature development and fixing as we go?

**Comment by hdefazio** [comment]
> Where do other remaining items here stand w.r.t. being technical debt as opposed to feature development and fixing as we go?

I think landing the voice demo would count as technical debt (ish)... and maybe cleaning up the python dependencies. The GSP can be debugged under the medical domain, so I wouldn't call that debt. Messing with the activity labels is specific to the cooking domain, so I'd leave it as technical debt since we won't circle back to fixing it for a while. 

But maybe there's a better distinction between technical debt and wrapping up the cooking domain

**Comment by cameron-a-johnson** [comment]
> > Where do other remaining items here stand w.r.t. being technical debt as opposed to feature development and fixing as we go?
> 
> I think landing the voice demo would count as technical debt (ish)... and maybe cleaning up the python dependencies. The GSP can be debugged under the medical domain, so I wouldn't call that debt. Messing with the activity labels is specific to the cooking domain, so I'd leave it as technical debt since we won't circle back to fixing it for a while.
> 
> But maybe there's a better distinction between technical debt and wrapping up the cooking domain

Fair point, Hannah - generally a category of technical debt might be "adaptability of the code base to new instruction domains," which our next steps will shed light on. Some of these changes have been very easy fixes, but some might be profound and better left off our lists for now (e.g. allowing for a non-linear set of steps).

**Comment by Purg** [comment]
Calling this done for now since things have progressed significantly since this.
